packages/kit/src/vite/index.js

```diff
@@ -320,12 +319,26 @@ function kit() {
 			process.env.SVELTEKIT_SERVER_BUILD_COMPLETED = 'true';
 			log.info('Prerendering');
 
-			prerendered = await prerender({
-				config: svelte_config.kit,
-				client_out_dir: vite_config.build.outDir,
-				manifest_path,
-				log
-			});
+			const results_path = `${svelte_config.kit.outDir}/generated/prerendered.json`;
+
+			// do prerendering in a subprocess so any dangling stuff gets killed upon completion
+			spawnSync(
+				'node',
+				[
+					'./node_modules/@sveltejs/kit/dist/prerender.js',
+					'--client_out_dir',
+					vite_config.build.outDir,
+					'--results_path',
+					results_path,
+					'--manifest_path',
+					manifest_path,
+					'--verbose',
+					'' + verbose
+				],
+				{ stdio: 'inherit' }
+			);
+
+			prerendered = JSON.parse(fs.readFileSync(results_path, 'utf8'));
```

Comment:
Do we need safeguards to ensure the file content is from the current run and there were no problems?

Reply:
ah, yeah, it was no longer failing the build if prerendering failed. fixed that